File: Phase-Gate-One/Python/subtraction_game.py
import random
import time

current_time = time.time()

questions = 10
attempts = 2
score = 0

print("Welcome to Subtraction Game")

def subtract(num1, num2):
    if num1 >= num2:
        return num1 - num2
    else:
        return num2 - num1

for index in range(1, 11):
    number1 = random.randint(0, 21)
    number2 = random.randint(0, 21)

    # Ensure number1 >= number2
    if number1 < number2:
        number1, number2 = number2, number1

    print(f"\nQuestion {index}: What is {number1} - {number2}?")

    while True:
        try:
            user_input = int(input("Your answer: "))
            break
        except:
            print("Invalid input! Enter a number.")

    if user_input == subtract(number1, number2):
        score += 1
        print("Correct! Well done.")
    else:
        attempts -= 1
        print("Wrong answer.")
        print("You have only", attempts, "attempt(s) left.")

        if attempts < 1:
            print("\nGAME OVER!")
            break

print("\nYour final score is:", score)

File: JavaScript/subtractiongame.js
    const prompt = require("prompt-sync")({sigint: true});
    let questions = 10;
    let questionAttempt = 2;
    let loss = 0;
    let score = 0;
    let StartTime = Date.now();


    
Console.log("Welcome to Subtraction Game");

for(let index = 1; index <= questions; index++){

            let number1 = Math.random(20)+1;
            let number2 = Math.random(20)+1;
            
              if(number1 < number2){
                number2 = number1;
              }

                if(attempt < 1){
                    System.out.println("Game over");
                    break;

                    }

            conslole.log(number1 + "-" + number2 +" = ");

            let userInput = input.nextInt();
                 
              if(userInput == subtract(number1, number2)){
                score++;

                console.log("Good boy, you get sharp brains");
                }
              else if(userInput != subtract(number1, number2)){
              attempt--;

              conslole.log("Dullard boy, you no sabi anything for this life, try again! ");
                
              console.log("You hava only " + attempt + " attempt left.");
              }
                
        }
let endTime = Date.now();

    

  function subtract(number1, number2){
    // switch place if number 1 is smaller than number 2
        let result = 0;
        let extra;
            if(number1 > number2){
            result+= number1 - number2;
            }
            if (number1 < number2){
            extra = number1;
            number1 = number2;
            result+= number1 - extra;
            }

            return result;
        

      }


